doc/sphinx/using/libraries/writing.rst

```diff
@@ -67,6 +67,43 @@ deprecated compatibility alias using :cmd:`Notation (abbreviation)`
 
    See :cite:`Zimmermann19`, Section 3.6.3, for more details.
 
+Triggering warning for library objects or library files
+-------------------------------------------------------
+
+You may use the following :term:`attribute` to trigger a warning on a
+notation, definition, axiom, theorem or file.
+
+.. attr:: warn ( note = @string , {? cats = @string } )
+   :name: warn
+
+   The :n:`note` field will be used as the warning message, and
+   :n:`cats` is a comma separated list of categories to be used in the
+   warning name and categories. Leading and trailing spaces in each
+   category are trimmed, whereas internal spaces are changed to
+   hyphens. If both :n:`note` and :n:`cats` are present, either one
+   may appear first and they must be separated by a comma.
+
+   This attribute is supported by the following commands:
+   :cmd:`Notation`, :cmd:`Infix`, :cmd:`Definition`, :cmd:`Theorem`,
+   and similar commands. To attach it to a compiled library file, use
+   :cmd:`Attributes`.
+
+   It can trigger the following warning:
+
+   .. warn:: @string__note
+
+      :n:`@string__note` is the note. It's common practice to start it
+      with a capital and end it with a dot.
```

Comment:
To the risk of being "pointilleux", I've been told that the American word is "period" (which Coq uses by convention) and the British word is "full stop" (but I'm not the expert).

Reply:
Indeed, let's go for period.